dali/operators/image/remap/cvcuda/warp_perspective.cc

```diff
@@ -0,0 +1,213 @@
+    .InputDox(1, "matrix_data", "1D TensorList of float",
+              "Transformation matrix data. Should be used to pass the GPU data. "
+              "For CPU data, the `matrix` argument should be used.")
+    .NumOutput(1)
+    .InputLayout(0, {"HW", "HWC", "FHWC", "CHW", "FCHW"})
+    .AddOptionalArg<float>("size",
+                           R"code(Output size, in pixels/points.
+
+The channel dimension should be excluded (for example, for RGB images,
+specify ``(480,640)``, not ``(480,640,3)``.
+)code",
+                           std::vector<float>({}), true)
+    .AddOptionalArg<float>("matrix",
+                           R"doc(
+  Perspective transform mapping of destination to source coordinates.
+  If `inverse_map` argument is set to true, the matrix is interpreted
+  as a source to destination coordinates mapping.
+
+It is equivalent to OpenCV's ``warpAffine`` operation with the ``inverse_map`` argument being
+analog to the ``WARP_INVERSE_MAP`` flag.
+
+.. note::
+  Instead of this argument, the operator can take a second positional input, in which
+  case the matrix can be placed on the GPU.)doc",
+                           std::vector<float>({}), true, true)
+    .AddOptionalArg("border_mode",
+                    "Border mode to be used when accessing elements outside input image.",
+                    "constant")
+    .AddOptionalArg("interp_type", "Interpolation method.", "nearest")
+    .AddOptionalArg<float>("fill_value",
+                           "Value used to fill areas that are outside the source image when the "
+                           "\"constant\" border_mode is chosen.",
+                           std::vector<float>({}))
+    .AddOptionalArg<bool>("inverse_map", "Inverse perspective transform matrix", false);
+
+
+class WarpPerspective : public nvcvop::NVCVSequenceOperator<StatelessOperator> {
+ public:
+  explicit WarpPerspective(const OpSpec &spec)
+      : nvcvop::NVCVSequenceOperator<StatelessOperator>(spec),
+        shape_arg_(spec.GetArgument<std::vector<float>>("size")),
+        border_mode_(nvcvop::GetBorderMode(spec.GetArgument<std::string>("border_mode"))),
+        interp_type_(nvcvop::GetInterpolationType(spec.GetArgument<std::string>("interp_type"))),
+        inverse_map_(spec.GetArgument<bool>("inverse_map")),
+        fill_value_arg_(spec.GetArgument<std::vector<float>>("fill_value")) {
+    matrix_data_.SetContiguity(BatchContiguity::Contiguous);
+  }
+
+  bool ShouldExpandChannels(int input_idx) const override {
+    return true;
+  }
+
+  bool CanInferOutputs() const override {
+    return true;
+  }
+
+  float4 GetFillValue(int channels) const {
+    if (fill_value_arg_.size() > 1) {
+      if (channels > 0) {
+        if (channels == static_cast<int>(fill_value_arg_.size())) {
+          float4 fill_value{0, 0, 0, 0};
+          memcpy(&fill_value, fill_value_arg_.data(), fill_value_arg_.size() * sizeof(float));
```

Comment:
paranoid nitpick (due to narrowing cast in the channels check):

```suggestion
          memcpy(&fill_value, fill_value_arg_.data(), sizeof(float4));
```

```suggestion
          memcpy(&fill_value, fill_value_arg_.data(), sizeof(decltype(fill_value)));
```